### PR TITLE
Patch Vulnerbility

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: helm/ffc-pr-calculator/Chart.yaml [hunk 1]
```diff
@@ -5,5 +5,5 @@ name: ffc-pr-calculator
 version: 1.0.0
 dependencies:
 - name: ffc-helm-library
-  version: ^3.1.0
```
Comment (johnwatson484) [inline helm/ffc-pr-calculator/Chart.yaml:8]: Updating to latest version of Helm library

File: helm/ffc-pr-calculator/values.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-environment: development
```
Comment (johnwatson484) [inline helm/ffc-pr-calculator/values.yaml:1]: This value is always overridden to `production` so no value in initially setting it to dev.  Reduces the amount of config we need by doing this.

File: helm/ffc-pr-calculator/values.yaml [hunk 3]
```diff
@@ -32,7 +27,6 @@ deployment:
   memoryThreshold: 80
 
 container:
-  imagePullPolicy: IfNotPresent
```
Comment (johnwatson484) [inline helm/ffc-pr-calculator/values.yaml:35]: These values are set by default in later versions of the helm library so no need to duplicate here

File: scripts/zap [hunk 4]
```diff
@@ -5,6 +5,6 @@ projectRoot="$(a="/$0"; a=${a%/*}; a=${a:-.}; a=${a#/}/; cd "$a/.." || return; p
 
 cd "${projectRoot}"
 # Guarantee clean environment
-docker-compose -f docker-compose.yaml -f docker-compose.zap.yaml -p ffc-demo-web-zap-scan down -v
+docker-compose -f docker-compose.yaml -f docker-compose.zap.yaml -p ffc-pr-calculator-zap-scan down -v
```
Comment (johnwatson484) [inline scripts/zap:8]: Clear copy and paste error that persisted throughout the ages!

File: webpack.config.js [hunk 5]
```diff
@@ -41,20 +41,16 @@ module.exports = {
...
       {
         test: /\.(png|svg|jpg|gif)$/,
-        use: {
-          loader: 'file-loader',
-          options: {
-            outputPath: 'images/'
-          }
+        type: 'asset/resource',
+        generator: {
+          filename: 'images/[name][ext]'
```
Comment (johnwatson484) [inline webpack.config.js:46]: These needed to be updated to work with later versions of Node.js

**Comment by AbidemiAdio** [comment]
I can see that package file has been modified to restrict packages to specific minor and patch versions by removing ^ and ~ in the main application folder, but these changes are not applied in the test folder package file. Is this intentional?

https://github.com/DEFRA/ffc-pr-calculator/blob/fix-vuln/test/acceptance/package.json

**Comment by johnwatson484** [comment]
> I can see that package file has been modified to restrict packages to specific minor and patch versions by removing ^ and ~ in the main application folder, but these changes are not applied in the test folder package file. Is this intentional?
> 
> https://github.com/DEFRA/ffc-pr-calculator/blob/fix-vuln/test/acceptance/package.json

Good point.  Wasn't intentional.  Now updated for consistency.

Related to this, I think we need to consider updating our UI tests at some point, or at least consider a slight variation on the approach for future services.  I feel there's a lot of opportunity to simplify and tidy up this setup so it's more maintainable.

**Review by AbidemiAdio — APPROVED** [review]
QED

**Pull request merged**